Tests/JWTKitTests/RSATests.swift

```diff
@@ -73,6 +73,57 @@ final class RSATests: XCTestCase {
         try await XCTAssertEqualAsync(await keyCollection.verify(privateSigned, as: TestPayload.self), payload)
     }
 
+    func testGetPublicKeyPrimitives() async throws {
+        let publicKey = try RSAKey(modulus: modulus, exponent: publicExponent)
+        let (keyModulus, exponent) = try publicKey.getPublicKeyPrimitives()
+        XCTAssertEqual(keyModulus, modulus)
+        XCTAssertEqual(exponent, publicExponent)
+    }
+
+    func testGetPrivateKeyPrimitives() async throws {
+        let privateKey = try RSAKey(modulus: modulus, exponent: publicExponent, privateExponent: privateExponent)
+        let (keyModulus, exponent, keyPrivateExponent) = try privateKey.getPrivateKeyPrimitives()
+        XCTAssertEqual(keyModulus, modulus)
+        XCTAssertEqual(exponent, publicExponent)
+        XCTAssertEqual(keyPrivateExponent, privateExponent)
+    }
+
+    func testGetPublicKeyPrimitivesWhenKeyIsPrivate() async throws {
+        let privateKey = try RSAKey(modulus: modulus, exponent: publicExponent, privateExponent: privateExponent)
+        let (keyModulus, exponent) = try privateKey.getPublicKeyPrimitives()
+        XCTAssertEqual(keyModulus, modulus)
+        XCTAssertEqual(exponent, publicExponent)
+    }
+
+    func testGetPrivateKeyPrimitivesWhenKeyIsPublicThrows() async throws {
+        let publicKey = try RSAKey(modulus: modulus, exponent: publicExponent)
+        XCTAssertThrowsError(try publicKey.getPrivateKeyPrimitives())
+    }
+
+    func testGetPrivateKeyPrimitivesFromNonRawBuiltKey() async throws {
+        let privateKey = try RSAKey.private(pem: """
```

Comment:
Can we also test that `BEGIN RSA PRIVATE KEY` will not decode as a PKCS#8 `PrivateKeyInfo` (or the public ones if that's easier) even if the data has the correct structure? (Conforming implementations are supposed to reject anything other than `BEGIN/END PRIVATE/PUBLIC KEY` for PKCS#8 PEM)